dataline-server/src/main/java/io/dataline/server/handlers/WebBackendConnectionsHandler.java

```diff
@@ -0,0 +1,106 @@
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package io.dataline.server.handlers;
+
+import io.dataline.api.model.ConnectionReadList;
+import io.dataline.api.model.JobConfigType;
+import io.dataline.api.model.JobListRequestBody;
+import io.dataline.api.model.JobReadList;
+import io.dataline.api.model.SourceImplementationIdRequestBody;
+import io.dataline.api.model.SourceImplementationRead;
+import io.dataline.api.model.WbConnectionRead;
+import io.dataline.api.model.WbConnectionReadList;
+import io.dataline.api.model.WorkspaceIdRequestBody;
+import io.dataline.commons.enums.Enums;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class WebBackendConnectionsHandler {
+
+  private final ConnectionsHandler connectionsHandler;
+  private final SourceImplementationsHandler sourceImplementationsHandler;
+  private final JobHistoryHandler jobHistoryHandler;
+
+  public WebBackendConnectionsHandler(
+                                      ConnectionsHandler connectionsHandler,
+                                      SourceImplementationsHandler sourceImplementationsHandler,
+                                      JobHistoryHandler jobHistoryHandler) {
+    this.connectionsHandler = connectionsHandler;
+    this.sourceImplementationsHandler = sourceImplementationsHandler;
+    this.jobHistoryHandler = jobHistoryHandler;
+  }
+
+  public WbConnectionReadList webBackendListConnectionsForWorkspace(
+                                                                    WorkspaceIdRequestBody workspaceIdRequestBody) {
+    final ConnectionReadList connectionReadList =
+        connectionsHandler.listConnectionsForWorkspace(workspaceIdRequestBody);
+
+    final List<WbConnectionRead> reads =
+        connectionReadList.getConnections().stream()
+            .map(
+                connectionRead -> {
+                  final SourceImplementationIdRequestBody sourceImplementationIdRequestBody =
+                      new SourceImplementationIdRequestBody();
+                  sourceImplementationIdRequestBody.setSourceImplementationId(
+                      connectionRead.getSourceImplementationId());
+                  final SourceImplementationRead sourceImplementation =
+                      sourceImplementationsHandler.getSourceImplementation(
+                          sourceImplementationIdRequestBody);
+
+                  final JobListRequestBody jobListRequestBody = new JobListRequestBody();
+                  jobListRequestBody.setConfigId(connectionRead.getConnectionId().toString());
+                  jobListRequestBody.setConfigType(JobConfigType.SYNC);
+                  final JobReadList jobReadList = jobHistoryHandler.listJobsFor(jobListRequestBody);
+
+                  final WbConnectionRead wbConnectionRead = new WbConnectionRead();
+                  wbConnectionRead.setConnectionId(connectionRead.getConnectionId());
```

Comment:
I have seen that pattern in a lot of places where we just copy an object to another. It is making me super nervous as the copy is not automated. the day we add a new field we will be in big trouble.

Reply:
do you have a suggestion or a different practice that you'd like me to follow?

my rule of thumb that i am using is if i am doing the conversion more than once i put in a helper function, otherwise i leave it like you see now.